### ISSUE TITLE
Potential issue with Take function handling unsigned integer

### ISSUE THREAD

**Issue by tigerwill90** [jub0bs/iterutil#1]
### Description:

The implementation of the `Take` function currently has a condition `count < 0`, which may lead to unintended behavior when count is an unsigned integer.

```go
seq := slices.Values([]string{"foo", "bar", "baz", "qux"})
for s := range iterutil.Take(seq, uint(2)) {
	fmt.Println(s)
}
// prints
// foo
// bar
// baz
// qux
```

If count is unsigned, the condition `count < 0` will never evaluate to `true` because unsigned integers cannot represent negative values. Instead, subtracting from 0 will cause count to wrap around to its maximum value.

###  Proposed Fix:
1. Constrain `I` to signed integers to ensure `count < 0` is a valid condition.
2. Alternatively, modify the condition to `count > 0` to handle both signed and unsigned integers.

```go
func Take[I constraints.Integer, E any](seq iter.Seq[E], count I) iter.Seq[E] {
	return func(yield func(E) bool) {
		for e := range seq {
			if count > 0 {
				if !yield(e) {
					return
				}
				count--
				continue
			}
			return
		}
	}
}
```

Hope I didn't overlook anything, and btw, thanks for this really useful package!

**Comment by jub0bs** [comment]
Glad to hear that you find the package useful.

Thanks for reporting this issue. Great catch! I think I will go with your second suggestion because I want to keep support unsigned integers (esp. `uint64`, in case of very long iterators). I don't have time to fix this right now, but I'll probably do it later this week.

Incidentally, I noticed a similar problem with `iterutil.Between`... I'll have to fix that as well.

**Comment by tigerwill90** [comment]
Hey @jub0bs, thanks for your response. Yes, I also think it's better to keep support for unsigned integers. By the way, I believe the same problem applies to `iterutil.Drop`, as `0 <= count` is always true for unsigned integers.

```go
values := slices.Values([]string{"foo", "bar", "baz"})

for s := range iterutil.Drop(values, 1) {
	fmt.Println(s) // prints: bar baz
}

for s := range iterutil.Drop(values, uint(1)) {
	fmt.Println(s) // prints nothing
}
```

Similarly, a possible solution is to change the condition to `count > 0` to handle both cases.

```go
func Drop[I constraints.Integer, E any](seq iter.Seq[E], count I) iter.Seq[E] {
	return func(yield func(E) bool) {
		for e := range seq {
			if count > 0 {
				count--
				continue
			}
			if !yield(e) {
				return
			}
		}
	}
}
```

**Comment by jub0bs** [comment]
@tigerwill90 Yes! I noticed it too as I was adding more tests for unsigned integers wherever `constraints.Integer` is used. I've already opened #2. But thanks for pointing out this issue with `Drop` anyway; I appreciate it. 🙇

**Comment by jub0bs** [comment]
Sorry @tigerwill90; it took me a while to get back to this, but it's now fixed in v0.5.0.
